crates/trie/parallel/src/proof_task.rs

```diff
@@ -40,73 +41,348 @@ use std::{
     time::Instant,
 };
 use tokio::runtime::Handle;
-use tracing::{debug, trace};
+use tracing::trace;
 
 #[cfg(feature = "metrics")]
 use crate::proof_task_metrics::ProofTaskMetrics;
 
 type StorageProofResult = Result<DecodedStorageMultiProof, ParallelStateRootError>;
 type TrieNodeProviderResult = Result<Option<RevealedNode>, SparseTrieError>;
 
-/// A task that manages sending multiproof requests to a number of tasks that have longer-running
-/// database transactions
+/// Internal message for storage workers.
+///
+/// This is NOT exposed publicly. External callers use `ProofTaskKind::StorageProof` or
+/// `ProofTaskKind::BlindedStorageNode` which are routed through the manager's `std::mpsc` channel.
+#[derive(Debug)]
+enum StorageWorkerJob {
+    /// Storage proof computation request
+    StorageProof {
+        /// Storage proof input parameters
+        input: StorageProofInput,
+        /// Channel to send result back to original caller
+        result_sender: Sender<StorageProofResult>,
+    },
+    /// Blinded storage node retrieval request
+    BlindedStorageNode {
+        /// Target account
+        account: B256,
+        /// Path to the storage node
+        path: Nibbles,
+        /// Channel to send result back to original caller
+        result_sender: Sender<TrieNodeProviderResult>,
+    },
+}
+
+impl StorageWorkerJob {
+    /// Sends an error back to the caller when worker pool is unavailable.
+    ///
+    /// Returns `Ok(())` if the error was sent successfully, or `Err(())` if the receiver was
+    /// dropped.
+    fn send_worker_unavailable_error(&self) -> Result<(), ()> {
+        match self {
+            Self::StorageProof { result_sender, .. } => {
+                let error = ParallelStateRootError::Other(
+                    "Storage proof worker pool unavailable".to_string(),
+                );
+                result_sender.send(Err(error)).map_err(|_| ())
+            }
+            Self::BlindedStorageNode { result_sender, .. } => {
+                let error = SparseTrieError::from(SparseTrieErrorKind::Other(Box::new(
+                    std::io::Error::new(
+                        std::io::ErrorKind::BrokenPipe,
+                        "Storage worker pool unavailable",
+                    ),
+                )));
+                result_sender.send(Err(error)).map_err(|_| ())
+            }
+        }
+    }
```

Comment:
can unify the match arms and send the same `ParallelStateRootError::Other` error

Reply:
sorry I meant that we can have one `Self::StorageProof { result_sender, .. } | Self::BlindedStorageNode { result_sender, .. }` arm, and send a `ParallelStateRootError::Other` on `result_sender`